### PR TITLE
Checking off the "Setup components" todo item. Mainly around the events list

### PR DESCRIPTION
**Pull request by antonjb**
In preparation for the "Pull Meetup events in from Meetup.com" item I
also moved events data to json and grabbing with GraphQL.

### CODE REVIEW

File: src/components/Footer.js [hunk 1]
```diff
@@ -1,26 +1,65 @@
...
+            <a className="logo" href="" title="Fenders">
+                <span>Fenders</span>
+            </a>
 
             <nav>
                 <ul>
-                    <li><a href="https://docs.google.com/forms/d/e/1FAIpQLScb2iYr65ZXUEDVkKP3uhbItojPKzHTOaBckjKUWrQoNLqSaA/viewform" title="Submit a talk">Submit a talk</a></li>
-                    <li><a href="mailto:fendersperth@gmail.com?Subject=Sponsorship" title="Sponsorship">Sponsorship</a></li>
-                    <li><a href="mailto:fendersperth@gmail.com?Subject=Enquiry" title="Email us">Email us</a></li>
+                    <li>
```
Comment (antonjb) [inline src/components/Footer.js]: Prettier...

File: src/data/events.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+	"events": [
+		{
+			"type": "social",
+			"title": "Casual Catchup",
+			"date": "2018-08-08T10:00:00.000Z",
```
Comment (antonjb) [inline src/data/events.json]: Store the dates as dates and get GraphQL to convert it to the format needed. Likely what will return from Meetup is something similar to this

File: src/data/events.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+			"url": "https://www.meetup.com/en-AU/Front-End-Web-Developers-Perth/events/qqvhjkyxlblb/",
+			"active": true
```
Comment (antonjb) [inline src/data/events.json]: It's possible this isn't needed and you could just add it to the first one I just wasn't sure what it was doing

Comment (amykapernick) [reply src/data/events.json]: @mandymichael may be able to confirm about this

File: src/data/events.json (hunk 2 above)
Comment (amykapernick) [reply src/data/events.json]: https://secure.meetup.com/meetup_api/console/?path=/:urlname/events

urlname = Front-End-Web-Developers-Perth

File: README.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 - The Fenders Perth Meetup website.
 
 ## TODO
-- [ ] Setup components
+- [x] Setup components
```
Comment (karlbright) [inline README.md]: Can just remove this if it's done?

File: README.md [hunk 5]
```diff
@@ -20,4 +20,4 @@ Get in touch with one of the organisers to get access to the repository :smiley:
 
 ## Getting approval access
 Want to help us on the website? We'd love you to help!
-Contact one of the Fenders Organisers and we can discuss adding you as a maintainer on the website 
\ No newline at end of file
+Contact one of the Fenders Organisers and we can discuss adding you as a maintainer on the website
```
Comment (karlbright) [inline README.md]: This should probably be done as part of another PR and most likely automatically done by your editor?

File: src/components/EventCard.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    return (
+        <article className={active ? 'active' : null}>
+            {month && <span className="month">{month}</span>}
+            {url ? <a href={url}>{cardContent}</a> : cardContent}
+        </article>
+    )
+}
+
+EventCard.propTypes = {
+    type: PropTypes.oneOf(['social', 'workshop', 'talk']),
```
Comment (karlbright) [inline src/components/EventCard.js]: This should probably be marked as `isRequired`?

File: src/components/EventCard.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import React, { Fragment } from 'react'
+import PropTypes from 'prop-types'
+
+const EventCard = ({ active, month, url, type, title, date }) => {
+    const cardContent = (
+        <Fragment>
+            <span
+                className={`tag ${type.toLowerCase()} icon`}
+            >{`${type[0].toUpperCase()}${type.substr(1)}`}</span>
```
Comment (karlbright) [inline src/components/EventCard.js]: This would not catch cases such as `SOCIAL`, which would result in the same string. Would it be easier to just bring in a package for titleize that can handle this? It also makes the code a lot easier to read.

File: src/components/Events.js [hunk 8]
```diff
@@ -1,37 +1,50 @@
...
-                <a href="https://www.meetup.com/en-AU/Front-End-Web-Developers-Perth/events/qqvhjkyxmbqb/" title="September Casual Catchup">
-                    <span className="tag social icon">Social</span>
-                    <strong>Casual Catchup</strong>
-                    <span className="date">Wednesday, September 12</span>
-                </a>
-            </article>
-        </div>
-    </section>
+                            return (
+                                <EventCard {...event} key={i} month={month} />
```
Comment (karlbright) [inline src/components/Events.js]: Should not use index for keys and try to use a unique identifier of some kind.

File: src/components/Events.js [hunk 9]
```diff
@@ -1,37 +1,50 @@
...
+                            )
+                        })}
+                    </div>
+                </section>
+            )
+        }}
+    />
 )
 
 // Event Types
```
Comment (karlbright) [inline src/components/Events.js]: Is this still required here? It's in the prop types.

File: src/components/Events.js [hunk 10]
```diff
@@ -1,37 +1,50 @@
...
+                        title
+                        date(formatString: "dddd, MMMM D")
+                        month: date(formatString: "MMMM")
+                        url
+                    }
+                }
+            }
+        `}
+        render={data => {
+            const usedMonths = []
```
Comment (karlbright) [inline src/components/Events.js]: I can't see where else this is used? The following code will push a momnth to the array if it is not previously added, but then the variable is not referenced anywhere else?

File: README.md (hunk 4 above)
Comment (antonjb) [reply README.md]: Unsure. It was setup as a checkbox so I assumed they wanted it ticked. @amykapernick ?

File: README.md (hunk 5 above)
Comment (antonjb) [reply README.md]: Didn't actually add that. I'm guessing VSCode added a new line automatically which highlighted it. Probably from above when I ticked the checkbox

File: src/components/EventCard.js (hunk 6 above)
Comment (antonjb) [reply src/components/EventCard.js]: Definitely should.

File: src/components/Events.js (hunk 8 above)
Comment (antonjb) [reply src/components/Events.js]: I was more setting this up to eventually grab from Meetup, which I assume would have an ID. I thought that graphql added ids but I can chuck something in the json file

File: src/components/Events.js (hunk 10 above)
Comment (antonjb) [reply src/components/Events.js]: Used here. The design only has the month the first time. So checks if it's there, then adds to here and continues on.
https://github.com/fendersperth/fenders-website/blob/e6614ff0c39ec896dbae87e54985cbe16be3bb95/src/components/Events.js#L29

I had hoped Graphql would be able to leave it out after the first instance but I couldn't see a way of doing that.

File: src/components/Events.js (hunk 9 above)
Comment (antonjb) [reply src/components/Events.js]: 🤷‍♂️  unsure. I avoided removing things when other people added the instructions but it is documented now in the proptypes

File: src/components/EventCard.js (hunk 7 above)
Comment (antonjb) [reply src/components/EventCard.js]: Would the oneof proptype catch case?

Comment (antonjb) [reply src/components/EventCard.js]: screen shot 2018-07-24 at 7 06 13 pm

Seems like it

**Comment by karlbright** [comment]
Excellent. Was wondering that myself. Nice one.
